lib/screens/pets/pet-profile/pet_profile.dart

```diff
@@ -0,0 +1,360 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:pet_app/model/pet.dart';
+import 'package:pet_app/model/user.dart';
+import 'package:pet_app/screens/home/home_page.dart';
+import 'package:pet_app/screens/pets/add-edit-pet/add-edit_pet_page.dart';
+import 'package:pet_app/screens/pets/my_pets/my_pets_page.dart';
+import 'package:pet_app/services/auth/auth_service.dart';
+import 'package:pet_app/services/pets/pets_service.dart';
+import 'package:pet_app/services/services.dart';
+import 'package:pet_app/services/user/user_service.dart';
+
+class PetProfile extends StatelessWidget {
+  static final String routeName = '/pet-profile';
+  final Pet displayedPet;
+
+  const PetProfile({Key key, @required this.displayedPet})
+      : assert(displayedPet != null),
+        super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final currentUserId = services
+        .get<AuthService>()
+        .currentUserUid;
+
+    return Scaffold(
+        resizeToAvoidBottomInset: false,
+        appBar: AppBar(
+          title: Text(displayedPet.name),
+          actions: currentUserId == displayedPet.ownerId
+              ? actionsForOwner(context)
+              : actionsForVisitor(),
+        ),
```

Comment:
This is one big Scaffold right here. Could it be reduced in size somehow?

Reply:
I've already reduced it a lot. Below there are some 6 functions which return some used widgets in the Scaffold

Reply:
I'll reduce it more...